### PR TITLE
CHUI 42: Integrate Tester Hub

### PR DESCRIPTION
**Pull request by wdsrocha**
#### What does this PR do?

Integrate [Tester Hub](https://github.com/vtex/tester-hub), making possible to run tests remotely.

#### How to test it?

1. Go to `checkoutio` account on `e2e` workspace
2. Run `vtex test e2e --token --workspace`

---

Adding a **#trivial** because [this](https://github.com/vtex/danger-plugin-keepachangelog/issues/1) keeps Danger from passing.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Added
 
 - Test profile on checkout step.
```
Comment (wdsrocha) [inline CHANGELOG.md:12]: I forgot to release this in the last version 🙃 . Is there a problem to release it like that now?

File: cypress/plugins/index.js [hunk 2]
```diff
@@ -1,23 +1,9 @@
...
-  // `on` is used to hook into various events Cypress emits
-  // `config` is the resolved Cypress config
+module.exports = (on, _config) => {
+  // eslint-disable-next-line default-param-last
+  on('before:browser:launch', (browser = {}, launchOptions) => {
+    if (browser.family === 'chromium' && browser.name !== 'electron') {
+      launchOptions.args.push('--disable-dev-shm-usage')
+    }
+    return launchOptions
+  })
```
Comment (wdsrocha) [inline cypress/plugins/index.js:8]: The plugin was added to resolve few crashes due to the lack of memory. The Tester Hub team is trying to solve this issue in a different way, but for now, it's recommended to keep it like that.

File: docs/README.md [hunk 3]
```diff
@@ -12,17 +12,24 @@ The repository responsable for all end-to-end tests on Checkout IO. If you are l
...
 
-To run ALL tests without sending to monitoring run:
+You can install those apps with the following command:
 
-```sh
-yarn e2e:run
 ```
+vtex install vtex.cypress-e2e-worker@0.0.4-0 &&
+vtex install vtex.tester-e2e-cypress@0.2.1-beta.0 &&
+vtex install vtex.tester-hub@0.3.1-beta.0
```
Comment (lucasecdb) [inline docs/README.md:27]: should these beta and pre-release versions be pinned here? shouldn't we use just `vtex install vtex.tester-hub`, etc, without the versioning?

Comment (wdsrocha) [reply docs/README.md:27]: I agree that this isn't ideal, but as far as I know, it still does not work outside beta. I'm going to check this.

Comment (wdsrocha) [reply docs/README.md:27]: Yeah @lucasecdb , it does not work outside beta (with exception of `tester-hub`). I think we can let this like that and when the worker/tester gets updated, I will update the README as well. What do you say?

Comment (lucasecdb) [reply docs/README.md:27]: sounds good to me

File: CHANGELOG.md (hunk 1 above)
Comment (lucasecdb) [reply CHANGELOG.md:12]: no, it's fine :-)

**Comment by lucasecdb** [comment]
@wdsrocha I couldn't get the tests running in my workspace :-(

In the workspace you provided in the description I received some 500 errors, but after some minutes toolbelt asked me to login again, and after I did the tests executed fine.

I installed the 3 apps in the "Developing" section of the readme but I'm getting this error (I don't know if it is related to tester-hub _per se_):

image

**Comment by wdsrocha** [comment]
@lucasecdb Oh, did you linked the app first? Tried here and it didn't worked as well, until I remembered to link.

**Comment by lucasecdb** [comment]
> @lucasecdb Oh, did you linked the app first? Tried here and it didn't worked as well, until I remembered to link.

oh, I didn't! sorry

**Comment by wdsrocha** [comment]
> oh, I didn't! sorry

No problem, I'll add that to the README, but I believe that toolbelt should catch this error somehow.

I tried to make a fix on toolbelt and now the error is

`10:45:39.431 - error: Raw error: App "vtex.checkout-io-tests@0.0.1" was not found in the current workspace! Error: App "vtex.checkout-io-tests@0.0.1" was not found in the current workspace!`

Do you think that would be helpful to realize that you had to link beforehand? If yes, I'll open a PR there.

**Comment by lucasecdb** [comment]
> Do you think that would be helpful to realize that you had to link beforehand? If yes, I'll open a PR there.

absolutely! would be a lot better and faster to identify the issue

**Pull request merged**